app/full_text_search.py

```diff
@@ -170,21 +215,35 @@ def create_document(record_id, repo, **kwargs):
     fields.append(appengine_search.TextField(name='repo', value=repo))
     fields.append(appengine_search.TextField(name='record_id', value=record_id))
 
-    # Add name and location romanized by unidecode
-    for field in kwargs:
-        romanized_value = script_variant.romanize_word(kwargs[field])
-        fields.append(
-            appengine_search.TextField(name=field, value=romanized_value))
+    # Add name romanized by unidecode
+    fields.extend(create_romanized_name_fields(
+        script_variant.romanize_word,
+        given_name=kwargs['given_name'],
+        family_name=kwargs['family_name'],
+        full_name=kwargs['full_name'],
+        alternate_names=kwargs['alternate_names']))
+
+    # Add location romanized by unidecode
+    fields.extend(create_romanized_location_fields(
+        script_variant.romanize_word,
```

Comment:
"romanize_word" sounds vague in contrast to romanize_japanese_name_by_name_dict etc. Can you rename it to something like "romanize_word_by_unidecode"?